benchmark/src/DriverBench/ParallelBench.php

```diff
@@ -0,0 +1,199 @@
+final class ParallelBench
+{
+    /** @var string[] */
+    private static array $files = [];
+
+    public static function beforeClass(): void
+    {
+        // Generate files
+        $fileContents = str_repeat(file_get_contents(Data::LDJSON_FILE_PATH), 5_000);
+        foreach (self::getFileNames() as $file) {
+            file_put_contents($file, $fileContents);
+        }
+    }
+
+    public static function afterClass(): void
+    {
+        foreach (self::getFileNames() as $file) {
+            unlink($file);
+        }
+
+        self::$files = [];
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using single thread
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[Revs(1)]
+    public function benchMultiFileImport(): void
+    {
+        $collection = Utils::getCollection();
+        foreach (self::getFileNames() as $file) {
+            self::importFile($file, $collection);
+        }
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using multiple forked threads
+     *
+     * @param array{processes:int, files:string[], batchSize:int} $params
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[ParamProviders(['provideProcessesParameter'])]
+    #[Revs(1)]
+    public function benchMultiFileImportFork(array $params): void
+    {
+        $pids = [];
+        foreach (self::getFileNames() as $file) {
+            // Wait for a child process to finish if we have reached the maximum number of processes
+            if (count($pids) >= $params['processes']) {
+                $pid = pcntl_waitpid(-1, $status);
+                unset($pids[$pid]);
+            }
+
+            $pid = pcntl_fork();
+            if ($pid === 0) {
+                // If we reset, we can garantee that we get a new manager in the child process
+                // If we don't reset, we will get the same manager client_zval in the child process
+                // and share the libmongoc client.
+                Utils::reset();
+                self::importFile($file, Utils::getCollection());
+
+                // Exit the child process
+                exit(0);
+            }
+
+            if ($pid === -1) {
+                throw new RuntimeException('Failed to fork');
+            }
+
+            // Keep the forked process id to wait for it later
+            $pids[$pid] = true;
+        }
+
+        // Wait for all child processes to finish
+        while ($pids !== []) {
+            $pid = pcntl_waitpid(-1, $status);
+            unset($pids[$pid]);
+        }
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using amphp/parallel-functions with worker pool
+     *
+     * @param array{processes:int, files:string[], batchSize:int} $params
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[ParamProviders(['provideProcessesParameter'])]
+    #[Revs(1)]
+    public function benchMultiFileImportAmp(array $params): void
+    {
+        wait(parallelMap(
```

Comment:
This implementation uses a [worker process](https://github.com/amphp/parallel/blob/1.x/lib/Context/Process.php) that communicate with the parent process to receive serialized functions to call and send serialized results.